agent/Dockerfile

```
@@ -87,10 +87,32 @@ RUN CGO_ENABLED=0 GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -ldflags="-w -s
+# Stage: Agent base - validate target arch and capture package baseline
+# =============================================================================
+FROM ${AGENT_BASE_IMAGE} AS agent-base
+
+ARG TARGETARCH=amd64
+
+RUN case "${TARGETARCH}" in \
+      amd64|arm64) ;; \
+      *) echo "ERROR: Snapshot supports amd64 and arm64 agent images, not ${TARGETARCH}" >&2; exit 1 ;; \
+    esac
+
+RUN set -eu; \
```

Hadolint (2.15.1)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)